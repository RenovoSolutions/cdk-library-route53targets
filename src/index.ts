import {
  aws_route53 as route53,
} from 'aws-cdk-lib';

export interface LoadBalancerTargetAttributes {
  /**
   * The DNS name of the load balancer.
   */
  readonly dnsName: string;
  /**
   * The hosted zone ID of the load balancer.
   */
  readonly hostedZoneId: string;
}

export class LoadBalancerTargetFromAttributes implements route53.IAliasRecordTarget {
  constructor(private readonly loadBalancerTargetAttributes: LoadBalancerTargetAttributes) {}

  public bind(_record: route53.IRecordSet, _zone?: route53.IHostedZone): route53.AliasRecordTargetConfig {
    return {
      hostedZoneId: this.loadBalancerTargetAttributes.hostedZoneId,
      dnsName: `dualstack.${this.loadBalancerTargetAttributes.dnsName}`,
    };
  }
}
