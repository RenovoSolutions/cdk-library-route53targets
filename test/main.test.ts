import {
  App,
  Stack,
  aws_route53 as r53,
} from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { LoadBalancerTargetFromAttributes } from '../src/index';

test('Snapshot', () => {
  const app = new App();
  const stack = new Stack(app, 'TestStack', {
    env: {
      account: '123456789012', // not a real account
      region: 'us-east-1',
    },
  });

  const zone = new r53.HostedZone(stack, 'HostedZone', {
    zoneName: 'example.com',
  });

  new r53.ARecord(stack, 'AliasRecord', {
    zone,
    recordName: 'publiclb.example.com',
    target: r53.RecordTarget.fromAlias(new LoadBalancerTargetFromAttributes({
      dnsName: 'publiclb-1234567890.us-east-1.elb.amazonaws.com',
      hostedZoneId: 'Z3DZXE0Q79N41H',
    })),
  });

  expect(Template.fromStack(stack)).toMatchSnapshot();
});
