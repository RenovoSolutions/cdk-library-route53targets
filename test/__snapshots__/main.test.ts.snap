// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Snapshot 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "AliasRecord851000D2": {
      "Properties": {
        "AliasTarget": {
          "DNSName": "dualstack.publiclb-1234567890.us-east-1.elb.amazonaws.com",
          "HostedZoneId": "Z3DZXE0Q79N41H",
        },
        "HostedZoneId": {
          "Ref": "HostedZoneDB99F866",
        },
        "Name": "publiclb.example.com.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "HostedZoneDB99F866": {
      "Properties": {
        "Name": "example.com.",
      },
      "Type": "AWS::Route53::HostedZone",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
